src/plugins/tbin/tbinplugin.cpp

```diff
@@ -0,0 +1,358 @@
+                            ttile.tilesheet = cell.tile()->tileset()->name().toStdString();
+                            if ( cell.tile()->frames().size() == 0 )
+                            {
+                                ttile.staticData.tileIndex = cell.tile()->id();
+                            }
+                            else
+                            {
+
+                                // TODO: Check all frame durations are the same
+                                for ( Tiled::Frame frame : cell.tile()->frames() )
+                                {
+                                    ttile.animatedData.frameInterval = frame.duration;
+                                    tbin::Tile tframe;
+                                    tframe.tilesheet = ttile.tilesheet;
+                                    tframe.staticData.tileIndex = frame.tileId;
+                                    ttile.animatedData.frames.push_back( tframe );
+                                }
+                            }
+                        }
+                        tlayer.tiles.push_back( ttile );
+                    }
+                }
+                tiledToTbinProperties(layer, tlayer.props );
+                tmap.layers.push_back( std::move( tlayer ) );
+                tileLayerIdMap[ tmap.layers.back().id ] = &tmap.layers.back();
+            }
+            else
+            {
+                throw std::invalid_argument( "Only object and tile layers supported." );
+            }
+        }
+
+        for ( Tiled::ObjectGroup* objs : objGroups )
+        {
+            for ( Tiled::MapObject* obj : objs->objects() )
+            {
+               if ( obj->name() != "TileData" || obj->size().width() != 1 || obj->size().height() != 1 ||
+                    obj->position().x() != std::floor( obj->position().x() ) || obj->position().y() != std::floor( obj->position().y() ) )
+                   continue;
+
+                tbin::Layer* tiles = tileLayerIdMap[ objs->name().toStdString() ];
+                int idx = static_cast< int >( obj->position().x() + obj->position().y() * tiles->layerSize.x );
+                tiledToTbinProperties( obj, tiles->tiles[ idx ].props );
+            }
+        }
+
+        std::ofstream file( fileName.toStdString(), std::ios::trunc | std::ios::binary );
+        if ( !file )
+        {
+            mError = tr( "Could not open file for writing" );
+            return false;
+        }
+        tmap.saveToStream( file );
+        file.close();
+    }
+    catch ( std::exception& e )
+    {
+        mError = tr((std::string("Exception: ") + e.what()).c_str());
+        return false;
+    }
+
+    return true;
+}
+
+QString TbinMapFormat::nameFilter() const
+{
+    return tr("Tbin map files (*.tbin)");
+}
+
+QString TbinMapFormat::shortName() const
+{
+    return QLatin1String("tbin");
+}
+
+bool TbinMapFormat::supportsFile(const QString &fileName) const
+{
+    // ...
+
+    return false;
```

Comment:
This should probably return whether the extension of the file is `tbin`, and/or it could open the file and check for a certain expected first few bytes. That should allow Tiled to open .tbin files without explicitly selecting that file format in the open file dialog.